### DISCUSSION TITLE
1.19.0-rc.1

### DISCUSSION THREAD

**Discussion by david-lev** [david-lev/pywa#55] [Announcements]
## What's Changed
> Update with pip: `pip3 install -U pywa==1.19.0-rc.1`

- [async] adding beta support for async!

```python
from pywa_async import WhatsApp, types  # Import from `pywa_async` instead of `pywa`

wa = WhatsApp(...)

@wa.on_message(filters.media)
async def echo_media(_: WhatsApp, message: types.Message):
    await message.copy(to=message.sender)
```

**Full Changelog**: https://github.com/david-lev/pywa/compare/1.18.1...1.19.0-rc.1

<hr /><em>This discussion was created from the release <a href='https://github.com/david-lev/pywa/releases/tag/1.19.0-rc.1'>1.19.0-rc.1</a>.</em>

**Comment by dRAT3** [comment]
Hi man, using this version, for some reason it never registers the calback. No errors either. Do I need to await the creation of the `WhatsApp` or something?

**Reply by yehuda-lev** [reply]
Hi, thanks for reporting the error in handling.
At the moment you can set up webhook through the regulaer client (not async)
```py
from pywa import Whatsapp
```
 or through Facebook thanks

**Reply by dRAT3** [reply]
When we set it up through facebook, it's really dodgy, registration always works but then nothing comes in. I've had it happen on different servers and different domains. And then by some strike of luck it starts working and then we're good and it keeps working for that domain/phone/server..

**Reply by dRAT3** [reply]
Okay so the strike of luck is registering with pywa sync (not fb). Then switching back to async, then it works.

**Reply by yehuda-lev** [reply]
> Okay so the strike of luck is registering with pywa sync (not fb). Then switching back to async, then it works.

Update your PyWa to [1.20.2](https://github.com/david-lev/pywa/releases/tag/1.20.2) and it should work. 
> **You need to run the code with:**
`uvicorn wa:fastapi_app`
